### PR TITLE
libSMB2: new recipe

### PR DESCRIPTION
**Pull request by extrowerk**

### CODE REVIEW

File: dev-libs/libsmb2/libsmb2-3.0.0.recipe [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+	libsmb2${secondaryArchSuffix}_devel = $portVersion
+	devel:libsmb2$secondaryArchSuffix = $libVersionCompat
+	"
+REQUIRES_devel="
+	libsmb2$secondaryArchSuffix == $portVersion base
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libssl$secondaryArchSuffix
```
Comment (korli) [inline dev-libs/libsmb2/libsmb2-3.0.0.recipe:38]: Will libssl really used?

**Comment by Begasus** [comment]
Build fails on gcc2, so you can strike that :)
Aside from that some things need to be changed to get it installed on 32bit, got the changes here, but will probably be for tomorrow (other laptop) :)

**Comment by Begasus** [comment]
@extrowerk add the work I've done here for 32bit to https://github.com/Begasus/haikuports/commit/16d95c2dbbbbee1978fd0d933258af138d87a1bf
Could you have a look there and check on 64bit? (don't have the time to add commits here (yet)) :)

File: dev-libs/libsmb2/libsmb2-3.0.0.recipe (hunk 1 above)
Comment (Begasus) [reply dev-libs/libsmb2/libsmb2-3.0.0.recipe:38]: Not sure if it's used but configure checks for it (if I disable it I get, and even if it's disabled):
```
-- Could NOT find OpenSSL, try to set the path to OpenSSL root folder in the system variable OPENSSL_ROOT_DIR (missing: OPENSSL_CRYPTO_LIBRARY OPENSSL_INCLUDE_DIR) 
```

Comment (korli) [reply dev-libs/libsmb2/libsmb2-3.0.0.recipe:38]: asking because libssl doesn't come in REQUIRES.

**Comment by Begasus** [comment]
If things are OK I could bump this to version 4.0.0, builds ok with my local changes (as mentioned in an earlier comment)

**Comment by extrowerk** [comment]
Please push your changes here too, if possible, i don1t know when will i be able to work on this.

**Pull request merged**